src/lib/cfa/nsswitch.rb

```diff
@@ -0,0 +1,265 @@
+    private_constant :VENDOR_PATH
+
+    class << self
+      # Instantiates and loads the file
+      #
+      # This method is basically a shortcut to instantiate and load the content in just one call.
+      #
+      # @param file_handler [#read,#write] something able to read/write a string (like File)
+      # @return [Nsswitch] File with the already loaded content
+      def load(file_handler: Yast::TargetFile)
+        new(file_handler: file_handler).tap(&:load)
+      end
+    end
+
+    attr_reader :load_path, :write_path
+
+    # Constructor
+    #
+    # @param file_handler [#read,#write] something able to read/write a string (like File)
+    #
+    # @see CFA::BaseModel#initialize
+    def initialize(file_handler: Yast::TargetFile)
+      # Path for loading the content
+      @load_path = Yast::FileUtils.Exists(LOCAL_PATH) ? LOCAL_PATH : VENDOR_PATH
+      # Path for writing the content, always the local file
+      @write_path = LOCAL_PATH
+
+      super(AugeasParser.new("nsswitch.lns"), load_path, file_handler: file_handler)
+    end
+
+    # Loads the file content
+    #
+    # If the file does not exist, consider it as empty.
+    #
+    # @see CFA::BaseModel#load
+    def load
+      super
+      fix_collection_names(data)
+      @current_content = @parser.serialize(data)
+    rescue Errno::ENOENT # PATH does not exist yet
+      self.data = @parser.empty
+      @loaded = true
+    end
+
+    # Returns a list of defined databases
+    #
+    # @return [Array<String>] List of database names
+    def entries
+      databases.keys
+    end
+
+    # Service specifications for given database name
+    #
+    # @param db_name [String] the database name, e.g., "passwd" or "hosts"
+    #
+    # @return [Array<String>] database service specifications or empty list if db_name is not found
+    def services_for(db_name)
+      databases[db_name]&.services || []
+    end
+
+    # Update (or create if it does not exist yet) the entry for given database name
+    #
+    #
+    # @note Current services, if any, are completely replaced
+    #
+    # @see #entry_for
+    # @see DatabaseEntry#services=
+    #
+    # @param db_name [String] the database name, e.g., "passwd" or "hosts"
+    # @param services [Array<String>] service specifications for the database
+    def update_entry(db_name, services)
+      database = entry_for(db_name)
+      database.services = services
+    end
+
+    # Delete entry for given database name
+    #
+    # @param db_name [String] the database db_name, e.g., "passwd" or "hosts"
+    def delete_entry(db_name)
+      data.delete(database_matcher(db_name))
+    end
+
+    # Writes the current content to /etc/nsswitch.conf
+    #
+    # @see CFA::BaseModel#save
+    def save
+      return unless modified?
+
+      @parser.file_name = write_path if @parser.respond_to?(:file_name=)
+      @current_content = @parser.serialize(data)
+      @file_handler.write(write_path, @current_content)
```

Comment:
well, I still a bit do not like it as you here serialize twice. If you really want to avoid mtime modification, it can be done easier and more efficient with:
```
      @parser.file_name = write_path if @parser.respond_to?(:file_name=)
      content = @parser.serialize(data)
      if content != @current_content
        @file_handler.write(write_path, content)
        @current_content = content
      end
```
this way you do serialization just once.